app/uk/gov/hmrc/ngrloginregisterfrontend/controllers/ConfirmContactDetailsController.scala

```diff
@@ -23,37 +23,96 @@ import uk.gov.hmrc.domain.Nino
 
   def show(): Action[AnyContent] =
     authenticate.authWithUserDetails.async { implicit request =>
-      citizenDetailsConnector.getPersonDetails(Nino(request.nino.nino.getOrElse(""))).map {
-        case Left(error) => Status(error.code)(Json.toJson(error))
-        case Right(personDetails) => Ok(view(SummaryList(createSummaryRows(personDetails, request)), name(personDetails)))
+      val credId = CredId(request.credId.getOrElse(""))
+      val nino = Nino(request.nino.nino.getOrElse(""))
+      val email = Email(request.email.getOrElse(""))
+
+      connector.getRatepayer(credId).flatMap {
+        case Some(ratepayer) =>
+          val name = ratepayer.ratepayerRegistration.flatMap(_.name).map(_.value).getOrElse("")
+          Future.successful(Ok(view(SummaryList(createSummaryRowsFromRatePayer(ratepayer, request)), name)))
+
+        case None =>
+          citizenDetailsConnector.getPersonDetails(nino).flatMap {
+            case Left(error) =>
+              Future.successful(Status(error.code)(Json.toJson(error)))
+
+            case Right(personDetails) =>
+              val nameValue = name(personDetails)
+              val ratepayerRegistration = RatepayerRegistration(
+                name = Some(Name(nameValue)),
+                email = Some(email),
+                address = Some(buildAddress(personDetails))
+              )
+
+              val ratepayerData = RatepayerRegistrationValuation(credId, Some(ratepayerRegistration))
+
+              connector.upsertRatepayer(ratepayerData).map { _ =>
+                Ok(view(SummaryList(createSummaryRows(personDetails, request)), nameValue))
+              }
+          }
       }
     }
 
+  private def buildAddress(personDetails: PersonDetails): Address =
+    Address(
+      line1 = personDetails.address.line1.getOrElse(""),
+      line2 = personDetails.address.line2,
+      town = personDetails.address.line4.getOrElse(""),
+      county = personDetails.address.line5,
+      postcode = Postcode(personDetails.address.postcode.getOrElse("")),
+      country = personDetails.address.country.getOrElse("")
+    )
+
   def name(personDetails: PersonDetails): String = List(
     personDetails.person.firstName,
     personDetails.person.middleName,
     personDetails.person.lastName
   ).flatten.mkString(" ")
 
+  private[controllers] def createSummaryRowsFromRatePayer(ratepayerRegistrationValuation: RatepayerRegistrationValuation, request: AuthenticatedUserRequest[AnyContent])(implicit messages: Messages): Seq[SummaryListRow] = {
+      val address = ratepayerRegistrationValuation.ratepayerRegistration.flatMap(_.address).map { address => {
+        Seq(
+          address.line1,
+          address.line2.getOrElse(""),
+          address.postcode.value,
+          address.country
+        )
+      }}.getOrElse(Seq.empty)
+
+    val contactNumber = ratepayerRegistrationValuation.ratepayerRegistration.flatMap(_.contactNumber).map(number => number.value).getOrElse("")
+
+      Seq(
```

Comment:
I feel there's the possibility of a bit of code golf here to make things easier to read - could we have a utility util.UrlGet(NameController) that expands to Some(Link(Call("GET" etc.